exporter/opensearchexporter/README.md

```diff
@@ -0,0 +1,66 @@
+# OpenSearch Exporter
+
+| Status                   |           |
+| ------------------------ |-----------|
+| Stability                | [devel]   |
+| Supported pipeline types | traces    |
+| Distributions            | [contrib] |
+
+This exporter supports sending OpenTelemetry signals as documents to [OpenSearch](https://www.opensearch.org).
+
+The documents are sent using [observability catalog](https://github.com/opensearch-project/opensearch-catalog/tree/main/schema/observability) schema.
+
+## Configuration options
+
+- `endpoints`: List of OpenSearch URLs. If endpoints is missing, the
+  OPENSEARCH_URL environment variable will be used.
+- `num_workers` (optional): Number of workers publishing bulk requests concurrently.
+- `dataset` (default=`default`) a user-provided label. It is used to construct the name of the destination index or data stream.
+- `namespace` (default=`namespace`)a user-provided label. It is used to construct the name of the destination index or data stream.
+- `flush`: Event bulk buffer flush settings
+    - `bytes` (default=5242880): Write buffer flush limit.
+    - `interval` (default=30s): Write buffer time limit.
+- `retry`: Event retry settings
```

Comment:
This should probably use the [`exporterhelper.RetrySettings` type](https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/queued_retry.go#L239) to provide consistency of configuration across exporters.

Reply:
These values are used to create a backoff function for the OpenSearch client. 
We'll check if `exporterhelper.RetrySettings` can be used here.

Reply:
The config struct can be changed to use `exporterhelper.RetrySettings` but the implied behaviour may be different.

`exporterhelper.WithRetry` retries a batch of trace records, right? OpenSearch supports retrying one failed document from a bulk indexing request.

Would it be more appropriate to log documents that OpenSearch failed to index and move on? Currently they are added back to the bulk indexer queue to be tried later.

Reply:
You can use the [`consumererror.Traces` type](https://github.com/open-telemetry/opentelemetry-collector/blob/main/consumer/consumererror/signalerrors.go#L27-L29) to return an error containing the spans that failed to export.  The `exportererror` [error handler](https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/traces.go#L50-L56) will unwrap the error, extract the spans, and replace the request with one containing just the failed spans.